src/frontends/pytorch/src/op/index_put_.cpp

```diff
@@ -0,0 +1,101 @@
+    auto const_neg_1 = context.mark_node(v0::Constant::create(element::i32, Shape{}, {-1}));
+
+    auto input = context.get_input(0);
+    auto input_shape = context.mark_node(std::make_shared<v3::ShapeOf>(input, element::i32));
+    auto indices = context.get_input(1);
+    auto values = context.get_input(2);
+    auto accumulate = context.const_input<bool>(3);
+
+    auto indices_partial_shape = indices.get_partial_shape();
+    auto indices_first_dim = indices_partial_shape[0];
+    FRONT_END_OP_CONVERSION_CHECK(indices_first_dim.is_static(),
+                                  "We support only lists of tensors with v0::Constant number of elements.");
+    int64_t indices_list_len = indices_first_dim.get_length();
+    if (indices_list_len == 0) {
+        return {values};
+    }
+
+    auto const_indices_list_len = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {indices_list_len}));
+    auto split_indices = context.mark_node(std::make_shared<v1::Split>(indices, const_0, indices_list_len));
+
+    std::shared_ptr<ov::Node> broadcast_index_shape;
+    ov::Output<ov::Node> index;
+    if (indices_list_len > 1) {
+        index = split_indices->output(0);
+        for (int i = 1; i < indices_list_len; i++) {
+            index = context.mark_node(std::make_shared<v1::Add>(index, split_indices->output(i)));
+        }
+        broadcast_index_shape = context.mark_node(std::make_shared<v3::ShapeOf>(index, element::i32));
+        OutputVector indices_list;
+        for (int i = 0; i < indices_list_len; i++) {
+            auto broadcast =
+                context.mark_node(std::make_shared<v3::Broadcast>(split_indices->output(i), broadcast_index_shape));
+            auto unsqueeze = context.mark_node(std::make_shared<v0::Unsqueeze>(broadcast, const_neg_1));
+            indices_list.push_back(unsqueeze);
+        }
+        index = context.mark_node(std::make_shared<v0::Concat>(indices_list, -1));
+    } else {
+        index = split_indices->output(0);
+        broadcast_index_shape = context.mark_node(std::make_shared<v3::ShapeOf>(index, element::i32));
+        index = context.mark_node(std::make_shared<v0::Unsqueeze>(index, const_neg_1));
+    }
+
+    auto sub_data_shape =
+        context.mark_node(std::make_shared<v8::Slice>(input_shape, const_indices_list_len, const_max_int, const_1));
+    auto values_shape =
+        context.mark_node(std::make_shared<v0::Concat>(OutputVector{broadcast_index_shape, sub_data_shape}, 0));
+    values = context.mark_node(std::make_shared<v3::Broadcast>(values, values_shape));
+    values = context.mark_node(std::make_shared<v1::ConvertLike>(values, input));
+
+    std::shared_ptr<ov::Node> result;
+    if (accumulate) {
+        auto zeros = generate_zeros_with_convertlike(context, input_shape, input);
+        result = context.mark_node(std::make_shared<v3::ScatterNDUpdate>(zeros, index, values));
```

Comment:
Can `index` be negative?

Reply:
I general yes, but out `ScatterNDUpdate` from OV doesn't support this case.

Reply:
I added support for negative indices.